From d054a887eb4c4ddc4657c3951d7388b7f40b21bb Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Thu, 20 Mar 2025 06:42:06 +0000
Subject: [PATCH 4/7] i40e: Remove second parameter of __assign_str()

This commit accommodates commit 2c92ca849fcc("tracing/treewide: Remove
second parameter of __assign_str()") which removes second parameter of
 __assign_str(). This commit was merged in the v6.10 development cycle
and causes compilation failures with the v6.12 kernel, which are listed
below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=2c92ca849fcc

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/i40e_trace.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/i40e_trace.h b/src/i40e_trace.h
index af5cc1f..4562c67 100644
--- a/src/i40e_trace.h
+++ b/src/i40e_trace.h
@@ -98,7 +98,7 @@ DECLARE_EVENT_CLASS(
 		__entry->ring = ring;
 		__entry->desc = desc;
 		__entry->buf = buf;
-		__assign_str(devname, ring->netdev->name);
+		__assign_str(devname);
 	),
 
 	TP_printk(
@@ -143,7 +143,7 @@ DECLARE_EVENT_CLASS(
 		__entry->ring = ring;
 		__entry->desc = desc;
 		__entry->skb = skb;
-		__assign_str(devname, ring->netdev->name);
+		__assign_str(devname);
 	),
 
 	TP_printk(
@@ -185,7 +185,7 @@ DECLARE_EVENT_CLASS(
 	TP_fast_assign(
 		__entry->skb = skb;
 		__entry->ring = ring;
-		__assign_str(devname, ring->netdev->name);
+		__assign_str(devname);
 	),
 
 	TP_printk(
-- 
2.47.1

